### PR TITLE
fix:[CDS-73926]: convert runtime input yaml to jsonNode only if it's not empty

### PR DESCRIPTION
**Pull request by shalini-agr**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/50270)

### CODE REVIEW

**Comment by shalini-agr** [comment]
trigger uts

**Comment by shalini-agr** [comment]
trigger messagecheck

File: pipeline-service/service/src/main/java/io/harness/pms/triggers/TriggerExecutionHelper.java [hunk 1]
```diff
@@ -394,8 +396,8 @@ && isEmpty(triggerDetails.getNgTriggerConfigV2().getInputSetRefs())) {
         SecurityContextBuilder.setContext(new ServicePrincipal(PIPELINE_SERVICE.getServiceId()));
         switch (pipelineEntity.getHarnessVersion()) {
           case PipelineVersion.V0:
-            JsonNode yamlForValidatingSchema = executionHelper.getPipelineYamlWithUnResolvedTemplates(
-                YamlUtils.readAsJsonNode(runtimeInputYaml), pipelineEntity);
+            JsonNode yamlForValidatingSchema =
```
Comment (rathodmeetsatish) [inline pipeline-service/service/src/main/java/io/harness/pms/triggers/TriggerExecutionHelper.java:399]: please add ut

File: pipeline-service/service/src/main/java/io/harness/pms/triggers/TriggerExecutionHelper.java [hunk 2]
```diff
@@ -290,15 +290,17 @@ && isEmpty(triggerDetails.getNgTriggerConfigV2().getInputSetRefs())) {
...
         pipelineYaml = pipelineEntity.getYaml();
       } else {
+        runtimeInputJsonNode = YamlUtils.readAsJsonNode(runtimeInputYaml);
         String pipelineYamlBeforeMerge = pipelineEntity.getYaml();
         switch (pipelineEntity.getHarnessVersion()) {
           case PipelineVersion.V1:
             planExecutionMetadataBuilder.inputSetYaml(runtimeInputYaml);
             pipelineYaml = InputSetMergeHelperV1.mergeInputSetIntoPipelineYaml(
-                YamlUtils.readAsJsonNode(runtimeInputYaml), YamlUtils.readAsJsonNode(pipelineYamlBeforeMerge));
+                runtimeInputJsonNode, YamlUtils.readAsJsonNode(pipelineYamlBeforeMerge));
```
Comment (rathodmeetsatish) [inline pipeline-service/service/src/main/java/io/harness/pms/triggers/TriggerExecutionHelper.java:303]: please add uts

**Review by rathodmeetsatish — CHANGES_REQUESTED** [review]

**Review by archit-harness — APPROVED** [review]
> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/50270)* status: 1 file(s) to review. 1 change requests. 1 of 2 approvals obtained.

**Review by shalini-agr — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/50270)* status: 4 file(s) to review. 1 change requests. 2. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`pipeline-service/service/src/main/java/io/harness/pms/triggers/TriggerExecutionHelper.java` line 303 at r1](https://reviewable.io/reviews/harness/harness-core/50270#gh-1255508970:-NZjn_Yn1sgYFSWDQkZp:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/0576b9ff755697e781d3caca38a78c54799912a4/pipeline-service/service/src/main/java/io/harness/pms/triggers/TriggerExecutionHelper.java#L303)):*
<details><summary><i>Previously, rathodmeetsatish (Meet Rathod) wrote…</i></summary><blockquote>

please add uts
</blockquote></details>

Done.
___
*[`pipeline-service/service/src/main/java/io/harness/pms/triggers/TriggerExecutionHelper.java` line 399 at r1](https://reviewable.io/reviews/harness/harness-core/50270#gh-1255508519:-NZjn_jl8YxY5KP2clZq:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/0576b9ff755697e781d3caca38a78c54799912a4/pipeline-service/service/src/main/java/io/harness/pms/triggers/TriggerExecutionHelper.java#L399)):*
<details><summary><i>Previously, rathodmeetsatish (Meet Rathod) wrote…</i></summary><blockquote>

please add ut
</blockquote></details>

Done.

**Comment by shalini-agr** [comment]
trigger uts

**Review by rathodmeetsatish — APPROVED** [review]

**Comment by mohitg0795** [comment]
*[`pipeline-service/service/src/main/java/io/harness/pms/plan/execution/ExecutionHelper.java` line 217 at r3](https://reviewable.io/reviews/harness/harness-core/50270#-NZkSZK7CzfLiDdnXEpz:-NZkSZK7CzfLiDdnXEq-:b-55r5iu) ([raw file](https://github.com/harness/harness-core/blob/003d683499022f1c5f49c2755fffb6e3bcf54f91/pipeline-service/service/src/main/java/io/harness/pms/plan/execution/ExecutionHelper.java#L217)):*
> ```Java
>       boolean isDebug) {
>     JsonNode mergedRuntimeInputJsonNode = null;
>     if (!isEmpty(mergedRuntimeInputYaml)) {
> ```

Lets use "isNotEmpty()" here directly. Thats more intuitive and easier to read.

**Comment by shalini-agr** [comment]
trigger uts

**Review by shalini-agr — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/50270)* status: 6 file(s) to review. 3. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`pipeline-service/service/src/main/java/io/harness/pms/plan/execution/ExecutionHelper.java` line 217 at r3](https://reviewable.io/reviews/harness/harness-core/50270#-NZkSZK7CzfLiDdnXEpz:-NZkVjTeBBEAOptvwWs7:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/003d683499022f1c5f49c2755fffb6e3bcf54f91/pipeline-service/service/src/main/java/io/harness/pms/plan/execution/ExecutionHelper.java#L217)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

Lets use "isNotEmpty()" here directly. Thats more intuitive and easier to read.
</blockquote></details>

Done.

**Review by mohitg0795 — APPROVED** [review]
> Reviewed 4 of 4 files at r2, 1 of 2 files at r3, 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/50270)* status: 6 file(s) to review. 2. unresolved discussions. 1 of 2 approvals obtained.

**Review by archit-harness — APPROVED** [review]
> Reviewed 4 of 4 files at r2, 1 of 2 files at r3, 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/50270)* status: 2. unresolved discussions.

**Comment by shalini-agr** [comment]
trigger review

**Comment by shalini-agr** [comment]
trigger review

**Pull request merged**